dockerv2.go

```diff
@@ -0,0 +1,43 @@
+package txtdirect
+
+import (
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+var dockerRegexes = map[string]*regexp.Regexp{
+	"v2":        regexp.MustCompile("^\\/?v2\\/?$"),
+	"_catalog":  regexp.MustCompile("^/v2/_catalog$"),
+	"tags":      regexp.MustCompile("^/v2/(.*)/tags/(.*)"),
+	"manifests": regexp.MustCompile("^/v2/(.*)/manifests/(.*)"),
+	"blobs":     regexp.MustCompile("^/v2/(.*)/blobs/(.*)"),
+}
+
+var DockerRegex = regexp.MustCompile("^\\/v2\\/(.*\\/(tags|manifests|blobs)\\/.*|_catalog$)")
+
+func redirectDockerv2(w http.ResponseWriter, r *http.Request, rec record) error {
+	path := r.URL.Path
+	if dockerRegexes["v2"].MatchString(path) {
+		_, err := w.Write([]byte(http.StatusText(http.StatusOK)))
+		return err
+	}
+	if path != "/" {
+		regexType := DockerRegex.FindAllStringSubmatch(path, -1)[0]
+		if regexType[len(regexType)-1] == "" {
```

Comment:
I've added this check and replacement code because of a weird bug in ``DockerRegex``.
In ``_catalog`` case, it returns an empty second group but in other languages like javascript, there isn't any second group. (Actually, there shouldn't be a second group at all). No idea why it happens in go :smile:

Reply:
_catalog isn't used in the normal cases. It's hard to pinpoint, what's legacy and unused etc.
For now my tests have shown that _catalog and tags aren't really used, but let's keep them for now.